paymentsheet/src/test/java/com/stripe/android/customersheet/data/FakeCustomerSheetPaymentMethodDataSource.kt

```diff
@@ -0,0 +1,57 @@
+package com.stripe.android.customersheet.data
+
+import com.stripe.android.model.PaymentMethod
+import com.stripe.android.model.PaymentMethodFixtures.CARD_PAYMENT_METHOD
+import com.stripe.android.model.PaymentMethodUpdateParams
+
+private typealias AttachPaymentMethodOperation = (paymentMethodId: String) -> CustomerSheetDataResult<PaymentMethod>
+private typealias DetachPaymentMethodOperation = (paymentMethodId: String) -> CustomerSheetDataResult<PaymentMethod>
+private typealias UpdatePaymentMethodOperation =
+    (paymentMethodId: String, params: PaymentMethodUpdateParams) -> CustomerSheetDataResult<PaymentMethod>
+
+internal class FakeCustomerSheetPaymentMethodDataSource(
+    private val paymentMethods: CustomerSheetDataResult<List<PaymentMethod>> =
+        CustomerSheetDataResult.success(listOf(CARD_PAYMENT_METHOD)),
+    private val onAttachPaymentMethod: AttachPaymentMethodOperation = { id ->
+        CustomerSheetDataResult.success(findPaymentMethod(paymentMethods, id))
+    },
+    private val onDetachPaymentMethod: DetachPaymentMethodOperation = { id ->
+        CustomerSheetDataResult.success(findPaymentMethod(paymentMethods, id))
```

Comment:
For vertical mode, Jay and I made the default impls of fake functions a not implemented exception. This was useful for ensuring that functions were only called in the tests we expected them to be called in. Just an FYI in case you think this would be useful for customer sheet too!

Reply:
Yeah I think some of the tests were actually expecting values which is why I have them like this. I'll follow up by updating those tests and making these all `NotImplemented`